Code/Framework/AzCore/AzCore/Casting/numeric_cast.h

```diff
@@ -8,14 +8,25 @@
 
 #pragma once
 
+// This is disabled by default because it puts in costly runtime checking of casted values.
+// You can either change it here to enable it across the engine, or use push/pop_macro to enable per file/feature.
+// Note that if using push/pop_macro, you may get some of the functions not inline and the definition coming from
+// another compilation unit, in such case, you will have to push/pop_macro on that compilation unit as well.
+// #define AZ_NUMERICCAST_ENABLED 1
+
+#if AZ_NUMERICCAST_ENABLED != 1
```

Comment:
Shouldn't this check be `#if !AZ_NUMERICCAST_ENABLED`

Because if AZ_NUMERICCAST_ENABLED is defined it, but not set to a value would still pass this check.

Reply:
sure, will change